internal/ethapi/api.go

```diff
@@ -972,7 +972,11 @@ func (e *revertError) ErrorData() interface{} {
 // Note, this function doesn't make and changes in the state/blockchain and is
 // useful to execute and retrieve values.
 func (s *PublicBlockChainAPI) Call(ctx context.Context, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *StateOverride) (hexutil.Bytes, error) {
-	result, err := DoCall(ctx, s.b, args, blockNrOrHash, overrides, 5*time.Second, s.b.RPCGasCap())
+	timeout := s.b.HTTPRpcTimeout()
```

Comment:
Is this the only place where we use this? Seems odd, if we have a 'global' rpctimeout flag, to only ever apply it on the Call method?

Reply:
Going through internal/ethapi/api.go it seems to be the only rpc with that hard coded timeout. I'll dig in some more.

Reply:
There also this: https://github.com/ethereum/go-ethereum/blob/5441a8fa47249a9165ece43a2924efe2c7bcdba0/eth/backend.go#L325

A hard coded 5 minute timeout for filters api, eth_getLogs, eth_getFilterChanges etc. Seems incompatible with the 5 second timeout for eth_call.

Reply:
It's not the same kind of timeout -- filters generally live on a ws subscription, not a one-off http request/response pair.